### ISSUE TITLE
Regression in 0.8.x series: "Failed to interpret `cargo metadata`'s json"

### ISSUE THREAD

**Issue by cberner** [EmbarkStudios/cargo-deny#335]
**Describe the bug**
There's a regression between version 0.7.3 & 0.8.0. 0.7.3 works just fine, but 0.8.0 fails with the following error message:
```
2021-02-21 06:04:36 [ERROR] Failed to interpret `cargo metadata`'s json: the given version requirement is invalid at line 1 column 160046: the given version requirement is invalid at line 1 column 160046
```

**To Reproduce**
1. `git clone https://github.com/fleetfs/fleetfs && cd fleetfs`
2. `cargo install cargo-deny --force --version 0.7.3 --locked`
3. `cargo deny check licenses`
4. Observe that it passes
5. `cargo install cargo-deny --force --version 0.8.0 --locked`
6. `cargo deny check licenses`
7. Observe that it fails: `2021-02-21 06:04:36 [ERROR] Failed to interpret `cargo metadata`'s json: the given version requirement is invalid at line 1 column 160046: the given version requirement is invalid at line 1 column 160046`
8. I also tested with the latest 0.8.7, and that version fails as well.

**Expected behavior**
Passes successfully with version 0.8.0, just like version 0.7.3

**Device:**
 - OS: Linux

**Comment by cberner** [comment]
I was able to bisect it to this change: 8635dc58377b69179db60af538e119f63f5e467e

I'm not sure which part of that commit, but I would assume one of the dependencies changes caused this

**Comment by Jake-Shadle** [comment]
So this is actually a conflict between cargo and the version of semver used to parse versions and version requirements, [`raft v0.5.0`](https://crates.io/crates/raft/0.5.0) has a version requirement on `protobuf` of `~2.0-2.2` which can't be parsed by semver::VersionReq (cargo_metadata uses v0.11), but was parsed in 0.10, albeit it was actually incorrectly parsing it as if the `2.2` was a pre-release version.

If this problem was more common, I would probably look into patching cargo_metadata and making it more resilient to these kinds of version requirements, since cargo obviously _used_ to spit them out (not sure if that is still the case), but this is the first time I've ever seen a bug like this, so I think the best thing to do would be to propose a patch to `raft` for a 0.5.1 version that correctly specifies the protobuf version ('2' is a bit too loose IMO) and published it to crates.io so that cargo metadata emits a valid version requirement.

This issue is also resolved in the [0.6.0-alpha](https://crates.io/crates/raft/0.6.0-alpha) version, which was published a year and a half ago, so maybe they could just cut a non-alpha release as well?

Regardless, I think I am going to close this as semver is doing the correct thing, as that isn't a valid version requirement, but since crates.io is immutable that version requirement can't be changed for the version you currently rely on.

**Comment by cberner** [comment]
Ah, thanks! I'll follow up with the raft crate
